artifactory/utils/container/buildinfo.go

```diff
@@ -119,6 +133,135 @@ func setBuildProperties(buildName, buildNumber, project string, imageLayers []ut
+			return imageLayers, nil
+		}
+
+		// Filter layers to only include those from the default deployment repository
+		var filteredLayers []utils.ResultItem
+		for _, layer := range imageLayers {
+			if layer.Repo == repoConfig.DefaultDeploymentRepo {
+				filteredLayers = append(filteredLayers, layer)
+			}
+		}
+
+		if len(filteredLayers) > 0 {
+			log.Info("Filtered", len(imageLayers), "layers to", len(filteredLayers), "layers from default deployment repository:", repoConfig.DefaultDeploymentRepo)
+		} else {
+			log.Warn(fmt.Sprintf(`No layers found in default deployment repository '%s' for virtual repository '%s'.
+This may indicate that image layers exist in other repositories but not in the default deployment repository.
+Properties will not be set to maintain consistency with virtual repository configuration.
+To fix this, consider pushing the image directly to the virtual repository to ensure it lands in the default deployment repository.`, repoConfig.DefaultDeploymentRepo, originalRepo))
+			return []utils.ResultItem{}, nil
+		}
+
+		return filteredLayers, nil
+	}
+
+	// We know it's a virtual repository, so we need to get its default deployment repo
+	// We still need an API call here to get the DefaultDeploymentRepo, but only for virtual repos
+	log.Info("Making API call for virtual repository", originalRepo, "to get DefaultDeploymentRepo")
+	repoConfig, err := getRepositoryConfiguration(originalRepo, serviceManager)
+	if err != nil {
+		log.Debug("Failed to get repository configuration for virtual repository", originalRepo, ":", err.Error())
+		return imageLayers, err
+	}
+
+	// If it's a virtual repository but has no default deployment repo, return all layers
+	if repoConfig.DefaultDeploymentRepo == "" {
+		log.Debug("Virtual repository", originalRepo, "has no default deployment repository, proceeding with all layers")
+		return imageLayers, nil
+	}
+
+	// Filter layers to only include those from the default deployment repository
+	var filteredLayers []utils.ResultItem
+	for _, layer := range imageLayers {
+		if layer.Repo == repoConfig.DefaultDeploymentRepo {
+			filteredLayers = append(filteredLayers, layer)
+		}
+	}
+
+	if len(filteredLayers) > 0 {
+		log.Info("Filtered", len(imageLayers), "layers to", len(filteredLayers), "layers from default deployment repository:", repoConfig.DefaultDeploymentRepo)
+	} else {
+		log.Warn(fmt.Sprintf(`No layers found in default deployment repository '%s' for virtual repository '%s'. 
+This may indicate that image layers exist in other repositories but not in the default deployment repository. 	
+Properties will not be set to maintain consistency with virtual repository configuration. 
+To fix this, consider pushing the image directly to the virtual repository to ensure it lands in the default deployment repository.`, repoConfig.DefaultDeploymentRepo, originalRepo))
+		return []utils.ResultItem{}, nil
+	}
```

Comment:
```suggestion
	if len(filteredLayers) == 0 {
	    log.Warn(fmt.Sprintf(`No layers found in default deployment repository '%s' for virtual repository '%s'. 
This may indicate that image layers exist in other repositories but not in the default deployment repository. 	
Properties will not be set to maintain consistency with virtual repository configuration. 
To fix this, consider pushing the image directly to the virtual repository to ensure it lands in the default deployment repository.`, repoConfig.DefaultDeploymentRepo, originalRepo))
		return []utils.ResultItem{}, nil
	}
	log.Info("Filtered", len(imageLayers), "layers to", len(filteredLayers), "layers from default deployment repository:", repoConfig.DefaultDeploymentRepo)
```